### PR TITLE
Update bindings to raylib 4.0.0 and raygui 3.0

### PR DESCRIPTION
**Pull request by Willenbrink**
Hey!
This PR updates the bindings for both raylib and raygui to their latest releases.
To update the bindings I mainly deleted removed functions and renamed stuff. Nothing too interesting. I've also added some additional accessors for Color, Matrix, Shader and ShaderLoc.

### CODE REVIEW

**Comment by tjammer** [comment]
Nice, thanks! I'm a bit busy this weekend but will check the PR within the week

**Review by tjammer — CHANGES_REQUESTED** [review]
Looks good to me except for my two comments. I will go over the other PR once this is merged :)

File: src/raylib/raylib.mli [hunk 1]
```diff
@@ -647,6 +647,8 @@ and Matrix : sig
 
   type t = t' ctyp
 
+  val t : t Ctypes.typ
```
[the remark this answers is not in the document]
Comment (Willenbrink) [reply src/raylib/raylib.mli:650]: I used it for instanced rendering which requires a carray/cptr of matrices. I guess it's not strictly necessary with the workaround of #18 but I don't necessarily agree with removing it from the interface. Why not instead expose the ctyp for every type (or at least every type that occurs as a cptr argument in a raylib function)? I ran into a similar issue with ShaderLoc.t = int not being exposed when writing a shader.

File: src/raylib/raylib.mli [hunk 2]
```diff
@@ -2799,6 +2804,7 @@ val draw_billboard_pro :
   Texture.t ->
   Rectangle.t ->
   Vector3.t ->
+  Vector3.t ->
```
[the remark this answers is not in the document]
Comment (Willenbrink) [reply src/raylib/raylib.mli:2807]: Oops, thanks for pointing that out

File: src/raylib/raylib.mli (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjammer) [reply src/raylib/raylib.mli:650]: Yes, you're right. It's either that or expose the whole Ctypes type and get rid of the ctyp wrapper.

**Pull request merged**